### PR TITLE
Implement efficient settable quorum without library

### PR DESCRIPTION
**Pull request by jferas**
Implement settable quorum with small addition to governor bravo code

### CODE REVIEW

File: src/RadworksGovernor.sol [hunk 1]
```diff
@@ -174,11 +178,17 @@ contract RadworksGovernor is
...
-  function quorum(uint256) public pure override returns (uint256) {
-    return QUORUM;
+  function quorum(uint256) public view override returns (uint256) {
+    return _quorum;
+  }
+
+  /// @notice Set the amount of RAD required to meet the quorum threshold for a proposal.
+  /// @dev This function is only callable by executed governance proposals (thus, the Timelock).
+  function setQuorum(uint256 _newQuorum) public onlyGovernance {
+    emit QuorumSet(_newQuorum);
```
Comment (alexkeating) [inline src/RadworksGovernor.sol:190]: It may be good to check this event is emitted in the test.

File: src/RadworksGovernor.sol [hunk 2]
```diff
@@ -174,11 +178,17 @@ contract RadworksGovernor is
...
   /// @dev Our implementation ignores the block number parameter and returns a constant.
-  function quorum(uint256) public pure override returns (uint256) {
-    return QUORUM;
+  function quorum(uint256) public view override returns (uint256) {
+    return _quorum;
+  }
+
+  /// @notice Set the amount of RAD required to meet the quorum threshold for a proposal.
+  /// @dev This function is only callable by executed governance proposals (thus, the Timelock).
+  function setQuorum(uint256 _newQuorum) public onlyGovernance {
```
Comment (alexkeating) [inline src/RadworksGovernor.sol:189]: It would also be good to have a test that  doesn't use a proposal to call this method.

**Review by alexkeating — COMMENTED** [review]
Looks good, left some ideas for additional tests

**Review by wildmolasses — COMMENTED** [review]
agree w/ @alexkeating comments, otherwise looking good

File: src/RadworksGovernor.sol [hunk 3]
```diff
@@ -47,23 +51,23 @@ contract RadworksGovernor is
   /// @notice Human readable name of this Governor.
   string private constant GOVERNOR_NAME = "Radworks Governor Bravo";
 
-  /// @notice The number of RAD (in "wei") that must participate in a vote to meet quorum threshold.
-  uint256 private constant QUORUM = 4_000_000e18; // 4,000,000 RAD
-
   /// @param _initialVotingDelay The initial voting delay this Governor will enforce.
   /// @param _initialVotingPeriod The initial voting period this Governor will enforce.
   /// @param _initialProposalThreshold The initial number of RAD required to submit
-  /// a proposal this Governor will enforce.
```
[the remark this answers is not in the document]
Comment (jferas) [reply src/RadworksGovernor.sol:56]: Yes.. put it back: https://github.com/ScopeLift/radworks-governor-upgrade/pull/15/commits/b573aeb871b7c6696b280f999ed3b3faf171e0e8

File: src/RadworksGovernor.sol (hunk 1 above)
Comment (jferas) [reply src/RadworksGovernor.sol:191]: Done: 73f73c9c7237d16400c23d610d559f3ef7563e40

File: src/RadworksGovernor.sol (hunk 2 above)
Comment (jferas) [reply src/RadworksGovernor.sol:190]: Done: bdc426a8558ec5571abb46949b072e5dfdbfb81d

**Review by wildmolasses — APPROVED** [review]
lgtm

**Comment by jferas** [comment]
Closing without merge.. The `quorum(uint256)` method, which takes a block number and expects to get the quorum at that past block. If we don't checkpoint past quorum values, we can't return the past value. This could break clients/tooling that expect it to be correct for past blocks.

**Pull request closed without merging**